lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -735,83 +799,183 @@ sub get_dataset_response {
   return ($found, \@dataset_response);
 }
 
-# Returns a BeaconDatasetAlleleResponse for a
-# variant feature
+# Returns a BeaconDatasetAlleleResponse for a variant feature
 # Assumes that it exists
 sub get_dataset_allele_response {
-  my ($dataset, $assemblyId, $found, $vf, $sv, $variant_dt) = @_;
+  my ($dataset, $assemblyId, $found, $vfs, $sv, $variant_dt) = @_;
 
   my $dataset_id = $dataset->dbID();
   my $ds_response;
   # for now dataset error is null
   my $error;
 
-    $ds_response->{'datasetId'} = $dataset->short_name();
-    $ds_response->{'exists'} = undef;
-    $ds_response->{'error'} = undef;
-    $ds_response->{'frequency'} = undef;
-    $ds_response->{'callCount'} = undef;
-    $ds_response->{'sampleCount'} = undef;
-    $ds_response->{'note'} = undef;
-    $ds_response->{'externalUrl'} = undef;
-    $ds_response->{'info'} = undef;
+  $ds_response->{'id'} = $dataset->short_name();
+  $ds_response->{'exists'} = undef;
+  # $ds_response->{'error'} = undef;
+  $ds_response->{'resultsCount'} = undef;
```

Comment:
We can add value to resultsCount below, currently it shows _null_ .

Reply:
That makes sense. Updated!